# Project 1: Deaths by tuberculosis

by Paul King, 7 June 2016

This is the project notebook for Week 1 of The Open University's [_Learn to code for Data Analysis_](http://futurelearn.com/courses/learn-to-code) course.

In 2000, the United Nations set eight Millenium Development Goals (MDGs) to reduce poverty and diseases, improve gender equality and environmental sustainability, etc. Each goal is quantified and time-bound, to be achieved by the end of 2015. Goal 6 is to have halted and started reversing the spread of HIV, malaria and tuberculosis (TB).
TB doesn't make headlines like Ebola, SARS (severe acute respiratory syndrome) and other epidemics, but is far deadlier. For more information, see the World Health Organisation (WHO) page <http://www.who.int/gho/tb/en/>.

Given the population and number of deaths due to TB in some countries during one year, the following questions will be answered: 

- What is the total, maximum, minimum and average number of deaths in that year?
- Which countries have the most and the least deaths?
- What is the death rate (deaths per 100,000 inhabitants) for each country?
- Which countries have the lowest and highest death rate?

The death rate allows for a better comparison of countries with widely different population sizes.

## The data

The data consists of total population and total number of deaths due to TB (excluding HIV) in 2013 in each of the 28 EU countries. 

The data was taken in July 2015 from <http://apps.who.int/gho/data/node.main.POP107?lang=en> (population) and <http://apps.who.int/gho/data/node.main.593?lang=en> (deaths). The uncertainty bounds of the number of deaths were ignored.

The data was collected into an Excel file which should be in the same folder as this notebook.

In [40]:
import warnings
# warnings.simplefilter('ignore', FutureWarning)

from pandas import *
data = read_excel('WHO POP TB EU - PaulKing.xls')
data

Country  Population (1000s)  \
0                                             Austria                8495   
1                                             Belgium               11104   
2                                            Bulgaria                7223   
3                                             Croatia                4290   
4                                              Cyprus                1141   
5                                      Czech Republic               10702   
6                                             Denmark                5619   
7                                             Estonia                1287   
8                                             Finland                5426   
9                                              France               64291   
10                                            Germany               82727   
11                                             Greece               11128   
12                                            Hungary                9955   
13                                            Ireland                4627   
14                                              Italy               60990   
15                                             Latvia                2050   
16                                          Lithuania                3017   
17                                         Luxembourg                 530   
18                                              Malta                 429   
19                                        Netherlands               16759   
20                                             Poland               38217   
21                                           Portugal               10608   
22                                            Romania               21699   
23                                           Slovakia                5450   
24                                           Slovenia                2072   
25                                              Spain               46927   
26                                             Sweden                9571   
27  United Kingdom of Great Britain and Northern I...               63136   

    TB deaths  
0        29.0  
1        18.0  
2       150.0  
3        52.0  
4         2.3  
5        28.0  
6        24.0  
7        32.0  
8        17.0  
9       340.0  
10      300.0  
11       77.0  
12       81.0  
13       18.0  
14      310.0  
15       43.0  
16      250.0  
17        2.2  
18        1.5  
19       20.0  
20      650.0  
21      140.0  
22     1200.0  
23       35.0  
24       21.0  
25      240.0  
26       13.0  
27      340.0

## The range of the problem

The column of interest is the last one.

In [41]:
tbColumn = data['TB deaths']

The total number of deaths in 2013 is:

In [42]:
tbColumn.sum()

4434.0

The largest and smallest number of deaths in a single country are:

In [43]:
tbColumn.max()

1200.0

In [44]:
tbColumn.min()

1.5

From less than 2 to over a thousand deaths is a big range. The average number of deaths, over all countries in the data, can give a better idea of the seriousness of the problem in each country.
The average can be computed as the mean or the median. Given the wide range of deaths, the median is probably a more sensible average measure.

In [45]:
tbColumn.mean()

158.35714285714286

In [46]:
tbColumn.median()

39.0

The median is far lower than the mean. This indicates that some of the countries had a very high number of TB deaths in 2013, pushing the value of the mean up.

## The most affected

To see the most affected countries, the table is sorted in ascending order by the last column, which puts those countries in the last rows.

In [47]:
data.sort('TB deaths')

Country  Population (1000s)  \
18                                              Malta                 429   
17                                         Luxembourg                 530   
4                                              Cyprus                1141   
26                                             Sweden                9571   
8                                             Finland                5426   
13                                            Ireland                4627   
1                                             Belgium               11104   
19                                        Netherlands               16759   
24                                           Slovenia                2072   
6                                             Denmark                5619   
5                                      Czech Republic               10702   
0                                             Austria                8495   
7                                             Estonia                1287   
23                                           Slovakia                5450   
15                                             Latvia                2050   
3                                             Croatia                4290   
11                                             Greece               11128   
12                                            Hungary                9955   
21                                           Portugal               10608   
2                                            Bulgaria                7223   
25                                              Spain               46927   
16                                          Lithuania                3017   
10                                            Germany               82727   
14                                              Italy               60990   
9                                              France               64291   
27  United Kingdom of Great Britain and Northern I...               63136   
20                                             Poland               38217   
22                                            Romania               21699   

    TB deaths  
18        1.5  
17        2.2  
4         2.3  
26       13.0  
8        17.0  
13       18.0  
1        18.0  
19       20.0  
24       21.0  
6        24.0  
5        28.0  
0        29.0  
7        32.0  
23       35.0  
15       43.0  
3        52.0  
11       77.0  
12       81.0  
21      140.0  
2       150.0  
25      240.0  
16      250.0  
10      300.0  
14      310.0  
9       340.0  
27      340.0  
20      650.0  
22     1200.0

The table raises the possibility that a large number of deaths may be partly due to a large population. To compare the countries on an equal footing, the death rate per 100,000 inhabitants is computed.

In [48]:
populationColumn = data['Population (1000s)']
data['TB deaths (per 100,000)'] = tbColumn * 100 / populationColumn
data.sort('TB deaths (per 100,000)')

Country  Population (1000s)  \
19                                        Netherlands               16759   
26                                             Sweden                9571   
1                                             Belgium               11104   
4                                              Cyprus                1141   
5                                      Czech Republic               10702   
8                                             Finland                5426   
0                                             Austria                8495   
18                                              Malta                 429   
10                                            Germany               82727   
13                                            Ireland                4627   
17                                         Luxembourg                 530   
6                                             Denmark                5619   
14                                              Italy               60990   
25                                              Spain               46927   
9                                              France               64291   
27  United Kingdom of Great Britain and Northern I...               63136   
23                                           Slovakia                5450   
11                                             Greece               11128   
12                                            Hungary                9955   
24                                           Slovenia                2072   
3                                             Croatia                4290   
21                                           Portugal               10608   
20                                             Poland               38217   
2                                            Bulgaria                7223   
15                                             Latvia                2050   
7                                             Estonia                1287   
22                                            Romania               21699   
16                                          Lithuania                3017   

    TB deaths  TB deaths (per 100,000)  
19       20.0                 0.119339  
26       13.0                 0.135827  
1        18.0                 0.162104  
4         2.3                 0.201578  
5        28.0                 0.261633  
8        17.0                 0.313306  
0        29.0                 0.341377  
18        1.5                 0.349650  
10      300.0                 0.362639  
13       18.0                 0.389021  
17        2.2                 0.415094  
6        24.0                 0.427122  
14      310.0                 0.508280  
25      240.0                 0.511433  
9       340.0                 0.528845  
27      340.0                 0.538520  
23       35.0                 0.642202  
11       77.0                 0.691948  
12       81.0                 0.813661  
24       21.0                 1.013514  
3        52.0                 1.212121  
21      140.0                 1.319759  
20      650.0                 1.700814  
2       150.0                 2.076699  
15       43.0                 2.097561  
7        32.0                 2.486402  
22     1200.0                 5.530209  
16      250.0                 8.286377

## Conclusions

The EU countries had a total of about 4400 deaths due to TB in 2013. The median shows that half of these coutries had fewer than 39 deaths. The much higher mean (over 158) indicates that some countries had a very high number. The least affected were Malta and Luxembourg, with number of deaths listed as 1.5 and 2.2 respectively (some data checking is clearly needed. The most affected were Poland and Romania with 650 and 1200 thousand deaths in a single year. However, taking the population size into account, the least affected were Netherlands and Sweden with less than 0.15 deaths per 100 thousand inhabitants, and the most affected were Romania and Lithuania with over 5.5 deaths per 100,000 inhabitants.

One should not forget that most values are estimates, and that the chosen countries are a small and unrepresentative sample of all the world's countries. Nevertheless, they convey the message that TB is still a significant cause of fatalities, and that there is a huge disparity between countries, with several being highly affected.